### ISSUE TITLE
server: trim HTTP OWS before validating standard MCP headers

### ISSUE THREAD

**Issue by mattzcarey** [modelcontextprotocol/typescript-sdk#2452]
## Problem

`createMcpHandler` rejects a valid modern request when a runtime exposes leading or trailing HTTP optional whitespace on `Mcp-Name` (or the other standard MCP headers) through `Headers.get()`.

In workerd, a programmatically constructed request with:

```ts
new Request(url, {
  headers: { "Mcp-Name": "  test_x_mcp_header  " }
})
```

can expose the trailing OWS to the handler. `validateStandardRequestHeaders()` compares that raw value to `params.name`, producing HTTP 400 / `HeaderMismatch` even though RFC 9110 §5.5 requires field parsing to exclude surrounding OWS.

This is observable in `@modelcontextprotocol/conformance@0.2.0-alpha.9` as the failing `ServerAcceptsWhitespaceHeaderValue` check in `http-header-validation` against `@modelcontextprotocol/server@2.0.0-beta.2` on workerd. Node's Fetch implementation normalizes the value first, so the existing server-level test passes there and does not catch the runtime difference.

## Expected

The inbound HTTP boundary should strip only RFC OWS (`SP` and `HTAB`) from `MCP-Protocol-Version`, `Mcp-Method`, and `Mcp-Name` before classification and cross-checking.

An actual name containing leading/trailing whitespace remains unambiguous because `encodeMcpParamValue()` Base64-wraps it.

## Proposed regression coverage

Add a transport-neutral classifier/standard-header test using raw `InboundHttpRequest` values padded with spaces/tabs, then normalize those three header fields before classification and validation.
